### PR TITLE
Add a creator function for Registry that allows a custom Logger

### PR DESCRIPTION
**Pull request by jasobrown**
Add a creator function for `Registry` that allows a custom `Logger`

Also, lower the logging of `Sending measurements` statements to `DEBUG`.

### CODE REVIEW

File: registry.go [hunk 1]
```diff
@@ -55,11 +55,15 @@ func NewRegistryConfiguredBy(filePath string) (*Registry, error) {
 }
 
 func NewRegistry(config *Config) *Registry {
+	return NewRegistryWithLogger(config, defaultLogger())
+}
+
+func NewRegistryWithLogger(config *Config, logger Logger) *Registry {
```
Comment (brharrington) [inline registry.go]: Would it be cleaner to have the logger function and related information come from the config object? It seems like it is just an aspect of how the registry is configured.

Comment (jasobrown) [reply registry.go]: Works for me. I'll knock that out shortly ...

**Comment by jasobrown** [comment]
... Ignore for now, found a few bugs locally ...

**Comment by jasobrown** [comment]
OK, tests passing now. I've moved `Registry.log` to `Config.Log` and fixed up the references. It seemed slightly cleaner to do that than have a `Logger` field in both `Config` and `Registry`; the trade off is needing to dereference `registry.config` every time you want to log something.

**Pull request merged**